### PR TITLE
[MINI-5616] Deeplink implementation update

### PR DESCRIPTION
**Pull request by munir-rakuten**
# Description
Update Deeplink implementation.

## Links
MINI-5616

# Checklist
- [x] I have read the [contributing guidelines](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md#sdk-development-learning-path) (if submitting a feature PR)
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by shohiebsense — APPROVED** [review]
LGTM

File: testapp/build.gradle [hunk 1]
```diff
@@ -106,8 +108,10 @@ android {
             resValue "string", "build_version", buildVersion
             resValue "string", "miniapp_sdk_version", project.version
             resValue "string", "appcenter_secret", prodManifestPlaceholders.appCenterSecret
-            resValue "string", "sslPublicKey", prodManifestPlaceholders.sslPublicKey
-            resValue "string", "sslPublicKeyBackup", prodManifestPlaceholders.sslPublicKeyBackup
+            resValue "string", "sslPublicKey", debugManifestPlaceholders.sslPublicKey
+            resValue "string", "sslPublicKeyBackup", debugManifestPlaceholders.sslPublicKeyBackup
+            resValue "string", "sslPublicKeyProd", prodManifestPlaceholders.sslPublicKey
+            resValue "string", "sslPublicKeyBackupProd", prodManifestPlaceholders.sslPublicKeyBackup
```
Comment (shohiebsense) [inline testapp/build.gradle:114]: would be better if you name this sslPublicKeyProdBackup instead

File: testapp/build.gradle [hunk 2]
```diff
@@ -126,6 +130,8 @@ android {
             initWith release
             resValue "string", "sslPublicKey", stagingManifestPlaceholders.sslPublicKey
             resValue "string", "sslPublicKeyBackup", stagingManifestPlaceholders.sslPublicKeyBackup
+            resValue "string", "sslPublicKeyProd", prodManifestPlaceholders.sslPublicKey
+            resValue "string", "sslPublicKeyBackupProd", prodManifestPlaceholders.sslPublicKeyBackup
```
Comment (shohiebsense) [inline testapp/build.gradle:134]: same comment as above

File: testapp/build.gradle [hunk 3]
```diff
@@ -143,6 +149,8 @@ android {
...
```
Comment (shohiebsense) [inline testapp/build.gradle:153]: same comment as above

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt [hunk 4]
```diff
@@ -117,4 +121,129 @@ class MiniAppDisplayViewModel constructor(
...
+                        is MiniAppNotFoundException ->
+                            _errorData.postValue(NO_MINI_APP_FOUND_ERROR)
+                        is MiniAppTooManyRequestsError ->
+                            _containTooManyRequestsError.postValue(true)
+                        else -> {
+                            _errorData.postValue(e.message)
+                        }
+                    }
+                }
+            }
```
Comment (shohiebsense) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt:163]: Wrap into a new function instead

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt [hunk 5]
```diff
@@ -117,4 +121,129 @@ class MiniAppDisplayViewModel constructor(
...
```
Comment (shohiebsense) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt:205]: same suggestion as above

**Review by shohiebsense — CHANGES_REQUESTED** [review]
Leave some comments, other than that Looks Good, well done

**Review by shohiebsense — APPROVED** [review]
looks good, well done

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt [hunk 6]
```diff
@@ -24,6 +25,9 @@ class MiniAppDisplayViewModel constructor(
     private val _isLoading = MutableLiveData<Boolean>()
     private val _containTooManyRequestsError = MutableLiveData<Boolean>()
 
+    private val NO_PUBLISHED_VERSION_ERROR = "No published version for the provided Mini App ID."
+    private val NO_MINI_APP_FOUND_ERROR = "No Mini App found for the provided Project ID."
```
Comment (rleojoseph) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayViewModel.kt:29]: Can we move this strings.xml file ?

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1: Just a small suggestion.

**Pull request merged**